R/bayesianpredictionperformance.R

```diff
@@ -0,0 +1,269 @@
+    pars$hp.tau.dist  <- "dhalft"
+    pars$hp.tau.mean  <- options[["priorTauTLocation"]]
+    pars$hp.tau.sigma <- options[["priorTauTScale"]]
+    pars$hp.tau.df    <- options[["priorTauTDf"]]
+    pars$hp.tau.min   <- options[["priorTauTMin"]]
+    pars$hp.tau.max   <- options[["priorTauTMax"]]
+  }
+
+  dataset <- .metamiscOmitNAs(dataset, options)
+
+  fit <- tryCatch(metamisc::valmeta(
+    measure    = options[["measure"]],
+    cstat      = if (options[["measure"]] == "cstat" && options[["inputMeasure"]] != "")              dataset[, options[["inputMeasure"]]],
+    cstat.se   = if (options[["measure"]] == "cstat" && options[["inputSE"]] != "")                   dataset[, options[["inputSE"]]],
+    cstat.cilb = if (options[["measure"]] == "cstat" && sum(unlist(options[["inputCI"]]) != "") == 2) dataset[, options[["inputCI"]][[1]][1]],
+    cstat.ciub = if (options[["measure"]] == "cstat" && sum(unlist(options[["inputCI"]]) != "") == 2) dataset[, options[["inputCI"]][[1]][2]],
+    OE         = if (options[["measure"]] == "OE" && options[["inputMeasure"]] != "")                 dataset[, options[["inputMeasure"]]],
+    OE.se      = if (options[["measure"]] == "OE" && options[["inputSE"]] != "")                      dataset[, options[["inputSE"]]],
+    OE.cilb    = if (options[["measure"]] == "OE" && sum(unlist(options[["inputCI"]]) != "") == 2)    dataset[, options[["inputCI"]][[1]][1]],
+    OE.ciub    = if (options[["measure"]] == "OE" && sum(unlist(options[["inputCI"]]) != "") == 2)    dataset[, options[["inputCI"]][[1]][2]],
+    N          = if (options[["inputN"]] != "")      dataset[, options[["inputN"]]],
+    O          = if (options[["inputO"]] != "")      dataset[, options[["inputO"]]],
+    E          = if (options[["inputE"]] != "")      dataset[, options[["inputE"]]],
+    slab       = if (options[["inputLabels"]] != "") dataset[, options[["inputLabels"]]],
+    method     = "BAYES",
+    pars       = pars,
+    adapt      = options[["adapt"]],
+    burnin     = options[["burnin"]],
+    sample     = options[["sample"]],
+    n.chains   = options[["chains"]]
+  ),error = function(e)e)
+
+  # error handling
+  if (any(class(fit) %in% c("simpleError", "error"))) {
+    .quitAnalysis(paste0("metamisc package failed with the following error: '", fit[["message"]], "'"))
+  }
+
+  model[["object"]] <- fit
+
+  return()
+}
+.metamiscPriorAndPosteriorPlot <- function(jaspResults, options, dataset, ready) {
+
+  if (!is.null(jaspResults[["priorAndPosteriorPlot"]]))
+    return()
+
+  imgHeight  <- 1000
+  imgWidth   <- 520
+
+  priorAndPosteriorPlot   <- createJaspPlot(title = gettext("Prior and posterior plot"), width = imgWidth, height = imgHeight)
+  priorAndPosteriorPlot$position <- 2.1
+  priorAndPosteriorPlot$dependOn(c("priorAndPosteriorPlot", c(.metamiscDependencies, .metamiscDependenciesBayesian)))
+  jaspResults[["priorAndPosteriorPlot"]] <- priorAndPosteriorPlot
+
+  if (!ready)
+    return()
```

Comment:
instead, you should change this to
```suggestion
  if (!ready || jaspResults[["summaryTable"]]$getError())
    return()
```
for all output elements.